x/ethbridge/keeper/keeper.go

```diff
@@ -0,0 +1,97 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/tendermint/tendermint/libs/log"
+
+	"github.com/cosmos/peggy/x/ethbridge/types"
+	"github.com/cosmos/peggy/x/oracle"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Keeper maintains the link to data storage and exposes getter/setter methods for the various parts of the state machine
+type Keeper struct {
+	cdc *codec.Codec // The wire codec for binary encoding/decoding.
+
+	supplyKeeper types.SupplyKeeper
+	oracleKeeper types.OracleKeeper
+	codespace    sdk.CodespaceType
+}
+
+// NewKeeper creates new instances of the oracle Keeper
+func NewKeeper(cdc *codec.Codec, supplyKeeper types.SupplyKeeper, oracleKeeper types.OracleKeeper, codespace sdk.CodespaceType) Keeper {
+	return Keeper{
+		cdc:          cdc,
+		supplyKeeper: supplyKeeper,
+		oracleKeeper: oracleKeeper,
+		codespace:    codespace,
+	}
+}
+
+// Logger returns a module-specific logger.
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
+}
+
+// Codespace returns the codespace
+func (k Keeper) Codespace() sdk.CodespaceType {
+	return k.codespace
+}
+
+func (k Keeper) ProcessClaim(ctx sdk.Context, claim types.EthBridgeClaim) (oracle.Status, sdk.Error) {
+	oracleClaim, err := types.CreateOracleClaimFromEthClaim(k.cdc, claim)
+	if err != nil {
+		return oracle.Status{}, types.ErrJSONMarshalling(k.Codespace())
```

Comment:
```suggestion
		return oracle.Status{}, err
```

Reply:
@fedekunze can you explain this change?

Reply:
I was suggesting that the error returned here should contain the message from `err` instead of returning `types.ErrJSONMarshalling(k.Codespace())` as the error